File: src/main/java/ru/ifmo/is/mfl/reviews/dto/ReviewWithoutMovieDto.java
package ru.ifmo.is.mfl.reviews.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;

import ru.ifmo.is.mfl.common.framework.dto.CrudDto;
import ru.ifmo.is.mfl.users.dto.UserDto;

import java.time.Instant;

@Data
@EqualsAndHashCode(callSuper = true)
public class ReviewWithoutMovieDto extends CrudDto {
  private int id;
  private UserDto user;
  private boolean visible;
  private String text;
  private int rating;
  private Instant date;
  private int viewedCounter;
}

File: client/src/interfaces/review/dto/ReviewWithoutMovieDto.ts
import { CrudDto } from '@/interfaces/crud/dto/CrudDto';
import { UserDto } from '@/interfaces/user/dto/UserDto';

export interface ReviewWithoutMovieDto extends CrudDto {
  user: UserDto;
  visible: boolean;
  text: string;
  rating: number; // 1-10
  date: Date;
  viewedCounter: number;
}
